### PR TITLE
storage e2e: replace mock driver, update images

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup
/kind failing-test

#### What this PR does / why we need it:

https://testgrid.kubernetes.io/sig-storage-csi-ci#canary-on-master is failing because the new snapshotter image gets patched in via https://github.com/kubernetes-csi/csi-release-tools/blob/a6a1a7979bf3ebc2bb10d0e33dd11ab281d6d39e/prow.sh#L911-L932 while the deployed RBAC rules are for the stable snapshotter release and therefore don't have the "patch" verb enabled.

The csi mock driver has been obsoleted and should not have been used anymore. That https://github.com/kubernetes/kubernetes/pull/101672 hadn't been merged was an oversight.

Besides that it is good practice to stay up-to-date, so all images get updated where possible.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/cc @xing-yang

### CODE REVIEW

File: test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver-snapshotter.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ spec:
       serviceAccountName: csi-mock
       containers:
         - name: csi-snapshotter
-          image: k8s.gcr.io/sig-storage/csi-snapshotter:v4.0.0
+          image: k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.1
```
Comment (xing-yang) [inline test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver-snapshotter.yaml:18]: Should this be using the v5.0.0-rc1 image?

Comment (pohly) [reply test/e2e/testing-manifests/storage-csi/mock/csi-mock-driver-snapshotter.yaml:18]: I don't think so. The goal is to be as close as possible to the released csi-driver-host-path deployment..

The canary jobs cover the upcoming releases.

**Pull request merged**